src/plugins/intel_cpu/src/nodes/multiclass_nms.cpp

```diff
@@ -199,195 +322,48 @@ void MultiClassNms::executeDynamicImpl(dnnl::stream strm) {
-            if (shared) {
-                selected_index = _flattened_index(box_info.batch_index, box_info.box_index, m_numBoxes);
-                selected_base[2] = boxes[selected_index * 4];
-                selected_base[3] = boxes[selected_index * 4 + 1];
-                selected_base[4] = boxes[selected_index * 4 + 2];
-                selected_base[5] = boxes[selected_index * 4 + 3];
+    if (attr_sort_result_across_batch_ && (attr_sort_result_ != SortResultType::NONE)) {
+        if (attr_sort_result_ == SortResultType::SCORE) {
+            parallel_sort_by_score_across_batch(rois);
+        } else if (attr_sort_result_ == SortResultType::CLASSID) {
+            parallel_sort_by_class_across_batch(rois);
+        }
+    } else {
+        if (attr_sort_result_ == SortResultType::SCORE || attr_sort_result_ == SortResultType::NONE) {
+            if (attr_keep_top_k_ < 0) {
+                parallel_sort_by_score(rois);
             } else {
-                int64_t offset = 0;
-                for (int64_t i = 0; i < box_info.batch_index; i++) {
-                    offset += roisnum[i];
-                }
-                // selected index from (M, C, 4)
-                selected_index = _flattened_index((offset + box_info.box_index),
-                                                box_info.class_index, m_numClasses);
-                int idx = box_info.class_index * boxesStrides[0] + offset * boxesStrides[1];
-                const float* curboxes = boxes + idx; // a slice of boxes of current class current image
-                selected_base[2] = curboxes[4 * box_info.box_index];
-                selected_base[3] = curboxes[4 * box_info.box_index + 1];
-                selected_base[4] = curboxes[4 * box_info.box_index + 2];
-                selected_base[5] = curboxes[4 * box_info.box_index + 3];
+                // already sorted
             }
-        }
-        // TODO [DS NMS]: remove when nodes from models where nms is not last node in model supports DS
-        if (!isDynamicNode()) {
-            std::fill_n(selected_outputs + (output_offset + real_boxes) * 6, (selectedBoxesNum_perBatch - real_boxes) * 6, -1.f);
-            std::fill_n(selected_indices + (output_offset + real_boxes), selectedBoxesNum_perBatch - real_boxes, -1);
-            output_offset += selectedBoxesNum_perBatch;
-            original_offset += real_boxes;
-        } else {
-            output_offset += real_boxes;
-            original_offset += real_boxes;
+        } else if (attr_sort_result_ == SortResultType::CLASSID) {
+            parallel_sort_by_class(rois);
         }
     }
-}
 
-bool MultiClassNms::created() const {
-    return getType() == Type::MulticlassNms;
+    fill_outputs(rois, boxes, roisnum);
 }
 
-float MultiClassNms::intersectionOverUnion(const float* boxesI, const float* boxesJ, const bool normalized) {
+float MultiClassNms::intersection_over_union(const float* boxesI, const float* boxesJ, const bool normalized) {
```

Comment:
Please, use the original style of name convention for this operator in camelCase

Reply:
done